jina/executors/decorators.py

```diff
@@ -437,19 +421,19 @@ def arg_wrapper(*args, **kwargs):
                 f'num_batch={num_batch}')
 
             # assume all datas have the same length
-            full_data_size = _get_size(args[_slice_on[0]])
+            full_data_size = _get_size(args[slice_on])
             total_size = _get_total_size(full_data_size, b_size, num_batch)
             final_result = []
-            yield_dict = [isinstance(args[i], Dict) for i in _slice_on]
-            data_iterators = [batch_iterator(_get_slice(args[i],total_size), b_size) for i in _slice_on]
+            yield_dict = [isinstance(args[slice_on + i], Dict) for i in range(0,num_data)]
+            data_iterators = [batch_iterator(_get_slice(args[slice_on + i],total_size), b_size) for i in range(0, num_data)]
 
             for batch in data_iterators[0]:
-                args[_slice_on[0]] = dict(batch) if yield_dict[0] else batch
-                for idx,slice_idx in enumerate(_slice_on[1:]):
-                    batch_idx = next(data_iterators[idx+1])
-                    if yield_dict[idx+1]:
+                args[slice_on] = dict(batch) if yield_dict[0] else batch
```

Comment:
why is `yield_dict` needed? what is batch returning?

Reply:
batch returns a list of tuples of (key, value)

Reply:
ok, I see. Maybe it is better to consider in `batch_iterator` the case where a `dict` is provided?

Reply:
It feels that until obtaining the `iterators` the code with `batching_multi_input` may be merged?